L/libqasm/build_tarballs.jl

```diff
@@ -0,0 +1,46 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "libqasm"
+version = v"0.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/QuTech-Delft/libqasm.git", "1f317077e2f2462cafec8385666d6e996340d2e8")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+mkdir -p libqasm/build
+cd libqasm/build/
+# rm /usr/share/cmake/Modules/Compiler/._*
+# rm /usr/share/cmake/Modules/CompilerId/._*
+# rm /usr/share/cmake/Modules/._*
+git submodule init
+git submodule update
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DLIBQASM_COMPAT=ON -DBUILD_SHARED_LIBS=ON -DLIBQASM_BUILD_PYTHON=OFF ..
+export PATH=$PATH:/workspace/srcdir/libqasm/build/src/cqasm/tree-gen/:/workspace/srcdir/libqasm/build/src/cqasm/func-gen/
```

Comment:
You can't do this, you're trying to run foreign executables.  You have to do a double pass with bootstrap, where you first build the executable for the host, then use _that_ one to compile the target one.

Reply:
Okay, is there any example of this?

Reply:
LLVM 😂 You basically have to build first the executables you need with the host cmake toolchain file, then you install in hostprefix (so that it's automatically in PATH), then you do the build of the package.

Seriously, the problem is that cmake doesn't allow you to specify both host and target toolchains, one has to deal this kind of bootstrap processes manually, which is rather annoying. Any other saner build system (autoconf, meson) allows bootstrapping because it supports double toolchains.

Reply:
Seems like I'm gonna spend some days patching... Every year I ask myself the same question: "Why the hell I was in love with C++ and CMake?" 🤦

I'm gonna be a couple of days out so I will return to this next week.

Thanks @giordano